components/org.wso2.carbon.identity.oauth2.grant.organizationswitch/src/main/java/org/wso2/carbon/identity/oauth2/grant/organizationswitch/OrganizationSwitchGrant.java

```diff
@@ -140,6 +146,11 @@ public boolean validateGrant(OAuthTokenReqMessageContext tokReqMsgCtx) throws Id
                 appResideOrgId.equals(authenticatedUser.getUserResidentOrganization())) {
             authenticatedUser.setUserResidentOrganization(null);
         }
+
+        /* The shared user's domain can be different from the original user's domain. Hence, resolve the correct user
+        store domain. */
+        resolveUserStoreDomain(authenticatedUser, accessingOrgId);
```

Comment:
for better readability, shall we return the relevant userstore from the private method and amend the authenticatedUser object in the main method

Reply:
Change the method name accordingly